### PR TITLE
Wire up zone ambientURL

### PR DESCRIPTION
**Pull request by zzmp**
Zone entities have a property, `ambientURL`, which now sets the ambient texture for the zone. It can be scripted through the zone, but not through the scene.

If `ambientURL` is set, it will be used. If `ambientURL` is unset, but the background is showing a skybox, the skybox is used. Otherwise, OLD_TOWN_SQUARE is used as a default. There is no longer access to the ambient light presets, although they are left in the code.

### CODE REVIEW

**Comment by zzmp** [comment]
#### Intent

Replace the ambient light settings (already removed) with a data-driven counterpart, such that the ambient texture can be specified through the zone.

#### Testing

ambientURL property

Ensure the ambientURL sets the ambient texture (lights) appropriately. Make sure the fallbacks, as described in the PR description, occur correctly.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 35988ad08a1ca16f5f0e7380579379184e592408 available at: https://gist.github.com/d6eee50045e6aa06703c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dc47f5c99b8ae36dc60b749099729c89414eab49 available at: https://gist.github.com/d6eee50045e6aa06703c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.dmg

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -260,11 +262,18 @@ void EntityTreeRenderer::applyZonePropertiesToScene(std::shared_ptr<ZoneEntityIt
     auto sceneLocation = sceneStage->getLocation();
     auto sceneTime = sceneStage->getTime();
     
+    // zzmp
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: Is this dead code?

File: libraries/model/src/model/Stage.cpp [hunk 2]
```diff
@@ -202,14 +202,14 @@ void SunSkyStage::setSunModelEnable(bool isEnabled) {
...
-}
-void SunSkyStage::setSunIntensity(float intensity) {
-    _sunLight->setIntensity(intensity);
-}
-void SunSkyStage::setSunAmbientIntensity(float intensity) {
-    _sunLight->setAmbientIntensity(intensity);
+void SunSkyStage::setSunAmbientSphere(const gpu::SHPointer& sphere) {
+    if (sphere) {
+        _sunLight->setAmbientSphere(*sphere);
+    }
```
Comment (ZappoMan) [inline libraries/model/src/model/Stage.cpp]: coding standard: `} else {` on same line

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp (hunk 1 above)
Comment (zzmp) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: Yes, I forgot to remove my personal reminders.

File: libraries/model/src/model/Stage.cpp (hunk 2 above)
Comment (zzmp) [inline libraries/model/src/model/Stage.cpp]: I finally found the VS setting to stop doing this to me. Nice catch.

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 3]
```diff
@@ -562,32 +562,13 @@ static void loadLightProgram(const char* vertSource, const char* fragSource, boo
...
-void DeferredLightingEffect::setGlobalLight(const glm::vec3& direction, const glm::vec3& diffuse, float intensity, float ambientIntensity) {
-    auto light = _allocatedLights.front();
-    light->setDirection(direction);
-    light->setColor(diffuse);
-    light->setIntensity(intensity);
-    light->setAmbientIntensity(ambientIntensity);
-}
-
-void DeferredLightingEffect::setGlobalSkybox(const model::SkyboxPointer& skybox) {
-    _skybox = skybox;
```
Comment (zzmp) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: This is still necessary for skybox reflection effects.

**Comment by zzmp** [comment]
setGlobalSkybox sets the DeferredLightingEffect::_skybox, but it is only ever used to retrieve the gpu::TexturePointer, same as is used to generate the SphericalHarmonics for the AmbientSphere.

Setting DNM while I amend this to continue setting the skybox cubemap. It is necessary for the global directional deferred light pass.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1817b8ef2f6cc8cdf3ac16cb6823b385a4ef951c available at: https://gist.github.com/d6eee50045e6aa06703c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20bfdba14bca46a6348d6fdfebf51c3c174f04de available at: https://gist.github.com/d6eee50045e6aa06703c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7186.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7.  (I used same image as shown in your zone properties setup for ambient)

Zone lighting only - no ambient set.

hifi-snap-by-omegaheron-on-2016-02-25_09-10-50

With ambient image set.

hifi-snap-by-omegaheron-on-2016-02-25_09-11-48

Notice coloration in this one showing it is getting lighting from ambient light image.

hifi-snap-by-omegaheron-on-2016-02-25_09-13-47

Removing ambient image returns zone to how is should look without and removing zone returns render to what I've come to recognize as default.  Works for me.

**Comment by OmegaHeron** [comment]
@zzmp - I think I missed a error case on this yesterday.  After seeing reports in forum about things rendering black/dark I poked around a bit more.  If you start Interface in an area with no zone set you see something like following;

hifi-snap-by-omegaheron-on-2016-02-26_11-38-36

If you start interface in an area with a zone then go to location shown above (or start in "bad spot" move to an area with zone and return) you see;

hifi-snap-by-omegaheron-on-2016-02-26_11-38-53

**Comment by OmegaHeron** [comment]
If you want to test reproducing issue do this.  Move to hifi://heron/avt - exit interface.  Restart interface and you should see the darkness.  Move to hifi://heron then return to hifi://heron/avt and you should see it lit more as expected.

**Comment by sethalves** [comment]
https://app.asana.com/0/26225263936266/94401121928054

**Comment by OmegaHeron** [comment]
And I duped (unintentionally) error report in 
https://app.asana.com/0/26225263936266/94863974763287

**Pull request merged**